lib/lotus/routing/parsing/json_parser.rb

```diff
@@ -3,13 +3,32 @@
 module Lotus
   module Routing
     module Parsing
+      # Json parsing error
+      # This is raised when the json parser fails to parse a json string.
+      #
+      # @since x.x.x
+      class BodyParsingError < ::StandardError
```

Comment:
This must be defined in `lib/lotus/routing/parsing/parser.rb`, as we need to make it available for custom parsers.